src/flint/fmpz.jl

```diff
@@ -759,20 +759,11 @@ end
 #
 ###############################################################################
 
-function ^(x::ZZRingElem, y::Union{Int, UInt, ZZRingElem})
-  if isone(x) || iszero(y)
-    one(x)
-  elseif x == -1
-    isodd(y) ? deepcopy(x) : one(x)
-  elseif y < 0
+function ^(x::IntegerUnionOrPtr, y::IntegerUnionOrPtr)
```

Comment:
```suggestion
function ^(x::ZZRingElem, y::IntegerUnion)
```
Otherwise you re-define `^(::Int, ::Int)` which is type piracy